Source/WebCore/PAL/pal/crypto/commoncrypto/CryptoDigestCommonCrypto.cpp

```diff
@@ -190,34 +175,32 @@ String CryptoDigest::toHexString()
     return String::fromUTF8(result.span());
 }
 
-std::optional<Vector<uint8_t>> CryptoDigest::computeHash(CryptoDigest::Algorithm algo, const Vector<uint8_t>& data, UseCryptoKit useCryptoKit)
+Vector<uint8_t> CryptoDigest::computeHash(CryptoDigest::Algorithm algo, std::span<const uint8_t> data)
 {
-#if HAVE(SWIFT_CPP_INTEROP)
-    if (useCryptoKit == UseCryptoKit::Yes) {
-        switch (algo) {
-        case CryptoDigest::Algorithm::SHA_1:
-            return PAL::Digest::sha1(data.span());
-        case CryptoDigest::Algorithm::SHA_256:
-            return PAL::Digest::sha256(data.span());
-        case CryptoDigest::Algorithm::SHA_384:
-            return PAL::Digest::sha384(data.span());
-        case CryptoDigest::Algorithm::SHA_512:
-            return PAL::Digest::sha512(data.span());
-        case CryptoDigest::Algorithm::SHA_224:
-            RELEASE_ASSERT_NOT_REACHED();
-            return std::nullopt;
-        }
-        return std::nullopt;
-    }
-#else
-    UNUSED_PARAM(useCryptoKit);
-#endif
-
     std::unique_ptr<CryptoDigest> digest = WTF::makeUnique<CryptoDigest>();
     ASSERT(digest->m_context);
     digest->m_context->algorithm = algo;
     digest->m_context->ccContext = createCryptoDigest(algo);
-    digest->addBytes(data.span());
+    digest->addBytes(data);
     return digest->computeHash();
 }
+#else
+
+Vector<uint8_t> CryptoDigest::computeHash(CryptoDigest::Algorithm algo, std::span<const uint8_t> data)
+{
+    switch (algo) {
+    case CryptoDigest::Algorithm::SHA_1:
+        return PAL::Digest::sha1(data);
+    case CryptoDigest::Algorithm::SHA_256:
+        return PAL::Digest::sha256(data);
+    case CryptoDigest::Algorithm::SHA_384:
+        return PAL::Digest::sha384(data);
+    case CryptoDigest::Algorithm::SHA_512:
+        return PAL::Digest::sha512(data);
+    }
+    RELEASE_ASSERT_NOT_REACHED("Unsupported HASH Algorithm");
+    return { };
```

Comment:
Continued from above, this seems like a case where we would previously return a `nullopt`, and now we're constructing an empty vector, which strikes me as unexpected.

Would it be acceptable to abort here?

Reply:
I just put that for the compiler to be happy but there is an assert that will crash. So yea I am aborting with a message.